sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CryptographicKeyShouldNotBeTooShort.cs

```diff
@@ -24,11 +32,468 @@ public void Method()
+        }
+
+        public void CompliantKeySizeSet()
+        {
+            var rsa1 = new RSACng();
+            rsa1.KeySize = 2048;
+
+            var rsa2 = new RSAOpenSsl();
+            rsa2.KeySize = 3072;
+        }
+
+        public void NoncompliantKeySizeSet()
+        {
+            var rsa1 = new RSACng();
+            rsa1.KeySize = 1024; // Noncompliant {{Use a key length of at least 2048 bits for RSA cipher algorithm.}}
+
+            var rsa3 = new RSACryptoServiceProvider(); // Noncompliant
+            rsa3.KeySize = 2048; // Noncompliant {{Use a key length of at least 2048 bits for RSA cipher algorithm. This assignment does not actually update the underlying key size.}}
+
+            var rsa4 = new RSAOpenSsl();
+            rsa4.KeySize = 512; // Noncompliant
+        }
+
+        public void CompliantCreate()
+        {
+            var rsa1 = RSA.Create();
+            var rsa2 = RSA.Create(2048);
+            var rsa3 = RSA.Create(3072);
+        }
+
+        public void NoncompliantCreate()
+        {
+            var rsa1 = RSA.Create(512); // Noncompliant {{Use a key length of at least 2048 bits for RSA cipher algorithm.}}
+            var rsa2 = RSA.Create(1024); // Noncompliant {{Use a key length of at least 2048 bits for RSA cipher algorithm.}}
+        }
+    }
+
+    public class SystemSecurityCryptographyEC
+    {
+        public void CompliantConstructors()
+        {
+            var ec1 = new ECDiffieHellmanCng(); // Compliant - default EC key size is 521
+            var ec2 = new ECDsaCng(); // Compliant - default EC key size is 521
+            var ec3 = new ECDsaOpenSsl(); // Compliant - default EC key size is 521
+
+            // Valid key sizes are 256, 384, and 521 bits.
+            var ec4 = new ECDiffieHellmanCng(256);
+            var ec5 = new ECDsaCng(384);
+            var ec6 = new ECDsaOpenSsl(521);
+            var ec7 = new ECDsaOpenSsl((IntPtr)128);
+
+            var ec8 = new ECDiffieHellmanCng(ECCurve.NamedCurves.brainpoolP384r1);
+            var ec9 = new ECDsaCng(ECCurve.NamedCurves.brainpoolP384t1);
+            var ec10 = new ECDsaOpenSsl(ECCurve.NamedCurves.brainpoolP512r1);
+        }
+
+        public void NonCompliantConstructors()
+        {
+            var ec1 = new ECDiffieHellmanCng(ECCurve.NamedCurves.brainpoolP192r1); // Noncompliant {{Use a key length of at least 224 bits for EC cipher algorithm.}}
+            var ec2 = new ECDsaCng(ECCurve.NamedCurves.brainpoolP160t1); // Noncompliant {{Use a key length of at least 224 bits for EC cipher algorithm.}}
+            var ec3 = new ECDsaOpenSsl(ECCurve.NamedCurves.brainpoolP192t1); // Noncompliant
+        }
+
+        public void CompliantKeySizeSet()
+        {
+            var ec1 = new ECDiffieHellmanCng();
+            ec1.KeySize = 512;
+            ec1.KeySize = 128; // OK - because this is not a valid key size for this object
```

Comment:
TBH, this is for implementation simplicity: it's easier for DSACryptoServiceProvider to raise an issue all the time (since it's always noncompliant), and here it's easier to never raise an issue (sinced it's can only be compliant). And since these are edge cases (invalid use of these constructors), I would not expect it to happen on real code anyways